src/components/career-form/career-file-upload.tsx

```diff
@@ -0,0 +1,238 @@
+import { Input, InputContainer } from '../form/controls';
+import React, { ReactNode, useCallback } from 'react';
+import { DropzoneRootProps, useDropzone } from 'react-dropzone';
+import styled from 'styled-components';
+import { theme } from '../layout/theme';
+import {
+  FieldErrors,
+  FieldValues,
+  UseFormClearErrors,
+  UseFormRegister,
+  UseFormSetError,
+  UseFormSetValue,
+  UseFormWatch,
+} from 'react-hook-form';
+import { GridItem } from '../grid/grid';
+import { FilePreview } from './career-file-preview';
+import { FormError } from './career-components';
+import { CareerSelect } from './career-select';
+import { FileTitleWrapper } from './career-file-title';
+
+interface FileUploadProps {
+  setValue: UseFormSetValue<FieldValues>;
+  setError: UseFormSetError<FieldValues>;
+  clearErrors: UseFormClearErrors<FieldValues>;
+  register: UseFormRegister<FieldValues>;
+  watch: UseFormWatch<FieldValues>;
+  name: string;
+  selectedFiles: FileList;
+  errors: FieldErrors;
+  children: ReactNode | ReactNode[];
+}
+
+const MAX_NUMBER = 3;
+const MAX_SIZE = 20000000; // maximal size per document of personio
+
+export const FileUpload = ({
+  setValue,
+  setError,
+  clearErrors,
+  name,
+  register,
+  selectedFiles,
+  errors,
+  children,
+  watch,
+}: FileUploadProps) => {
+  const onDrop = useCallback(
+    (droppedFiles) => {
+      // maximal number of file being dropped
+      const maximalOfFiles = selectedFiles
+        ? MAX_NUMBER - selectedFiles.length
+        : MAX_NUMBER;
```

Comment:
next time a if-else condition works wonder to keep the code readable. The ternary operator is send from hell in terms of readability 👹